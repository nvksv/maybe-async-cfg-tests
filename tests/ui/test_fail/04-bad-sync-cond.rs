#[maybe_async_cfg::maybe(
    sync(feature = "is_sync"), 
    async(not(feature = "is_sync")),
)]
async fn async_fn() -> bool {
    true
}

// bad sync condition
//#[maybe_async::test(unknown(feature="async", async_std::test))]
#[maybe_async_cfg::maybe(
    idents(async_fn(fn)),
    sync(unknown = "is_sync"), 
    async(not(feature = "is_sync")),
)]
async fn test_async_fn() {
    let res = async_fn().await;
    assert_eq!(res, true);
}

fn main() {

}