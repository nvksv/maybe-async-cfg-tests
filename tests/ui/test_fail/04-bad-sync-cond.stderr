error: Wrong name for name-value pair: unknown
  --> tests/ui/test_fail/04-bad-sync-cond.rs:13:10
   |
13 |     sync(unknown = "is_sync"),
   |          ^^^^^^^^^^^^^^^^^^^
